librosa/core/notation.py

```diff
@@ -124,15 +124,30 @@
 
 # Pre-compiled regular expressions for note and key parsing
 KEY_RE = re.compile(
-    r"^(?P<tonic>[A-Ga-g])" r"(?P<accidental>([#♯b!♭𝄪𝄫♮]*))" r":(?P<scale>(maj|min)(or)?)$"
+    r"^(?P<tonic>[A-Ga-g])"
+        r"(?P<accidental>[#♯𝄪b!♭𝄫]*)"
+        r":((?P<scale>(maj|min)(or)?)|(?P<mode>(ion|dor|phr(yg)?|lyd|mix(olyd)?|aeol?|locr?)(ian)?))$"
 )
+
 NOTE_RE = re.compile(
     r"^(?P<note>[A-Ga-g])"
     r"(?P<accidental>[#♯𝄪b!♭𝄫♮]*)"
     r"(?P<octave>[+-]?\d+)?"
     r"(?P<cents>[+-]\d+)?$"
 )
 
+MAJOR_DICT = {
+    'ion': {'C': 'C', 'D': 'D', 'E': 'E', 'F': 'F', 'G': 'G', 'A': 'A', 'B': 'B'},
+    'dor': {'C': 'B♭', 'D': 'C', 'E': 'D', 'F': 'E♭', 'G': 'F', 'A': 'G', 'B': 'A'},
+    'phr': {'C': 'A♭', 'D': 'B♭', 'E': 'C', 'F': 'D♭', 'G': 'E♭', 'A': 'F', 'B': 'G'},
+    'lyd': {'C': 'G', 'D': 'A', 'E': 'B', 'F': 'C', 'G': 'D', 'A': 'E', 'B': 'F♯'},
+    'mix': {'C': 'F', 'D': 'G', 'E': 'A', 'F': 'B♭', 'G': 'C', 'A': 'D', 'B': 'E'},
+    'aeo': {'C': 'E♭', 'D': 'F', 'E': 'G', 'F': 'A♭', 'G': 'B♭', 'A': 'C', 'B': 'D'},
+    'loc': {'C': 'D♭', 'D': 'E♭', 'E': 'F', 'F': 'G♭', 'G': 'A♭', 'A': 'B♭', 'B': 'C'}
+}
+
+OFFSET_DICT = dict([(y,x) for (x,y) in enumerate(['ion', 'dor', 'phr', 'lyd', 'mix', 'aeo', 'loc'])])
```

Comment:
```suggestion
OFFSET_DICT = {mode: offset for (offset, mode) in enumerate(['ion', 'dor', 'phr', 'lyd', 'mix', 'aeo', 'loc'])}
```

Reply:
Or I guess we could just make this explicit:
```suggestion
OFFSET_DICT = { "ion": 0, "dor": 1, "phr": 2, "lyd": 3, "mix": 4, "aeo": 5, "loc": 6 }
```
it's not like the modes are changing very often these days...

Reply:
Might as well keep it simple.